src/mappings/factory.ts

```diff
@@ -74,11 +85,13 @@ export function handleNewExchange(event: NewExchange): void {
     factory.totalAddLiquidity = BigInt.fromI32(0)
     factory.totalRemoveLiquidity = BigInt.fromI32(0)
     factory.exchangeHistoryEntityCount = BigInt.fromI32(0)
+    factory.uniswapHistoryEntityCount = BigInt.fromI32(0)
     factory.txCount = BigInt.fromI32(0)
   }
   factory.exchangeCount = factory.exchangeCount + 1
   factory.save()
 
+  // create new exchange with data from our hard coded list
```

Comment:
Hmm, it is not possible for us to load the contracts automatically? It this because of the bug where `decimals()` returns `bytes` instead of `uint256 ` ???????

Reply:
exactly, bytes are causing the issue. The issue is that we need to enforce one type on decimals, but not all uints